core/src/main/java/com/sequenceiq/cloudbreak/converter/v2/TemplateV2RequestToTemplateRequestConverter.java

```diff
@@ -38,10 +40,11 @@ public TemplateRequest convert(TemplateV2Request source) {
         } else if (source.getAzureParameters() != null) {
             template.getParameters().putAll(getConversionService().convert(source.getAzureParameters(), Map.class));
         } else if (source.getGcpTemlateParameters() != null) {
-            template.getParameters().putAll(getConversionService().convert(source.getGcpTemlateParameters(), Map.class));
+            GcpParameters gcpParameters = Optional.ofNullable(source.getGcpTemlateParameters()).orElse(new GcpParameters());
+            template.getSecretParameters().putAll(getConversionService().convert(gcpParameters, Map.class));
```

Comment:
json or map is better?

Reply:
It seems to be a convention, that we use a map for parameters. Also map has better support for handling non-existing key-value pairs.